test/core/end2end/tests/hpack_size.cc

```diff
@@ -386,11 +392,22 @@ void hpack_size(grpc_end2end_test_config config) {
                                           1000, 32768, 4 * 1024 * 1024};
   size_t i, j;
 
+  std::vector<std::shared_ptr<absl::Notification>> dones;
   for (i = 0; i < GPR_ARRAY_SIZE(interesting_sizes); i++) {
     for (j = 0; j < GPR_ARRAY_SIZE(interesting_sizes); j++) {
-      test_size(config, interesting_sizes[i], interesting_sizes[j]);
+      auto done = std::make_shared<absl::Notification>();
+      dones.push_back(done);
+      grpc_event_engine::experimental::GetDefaultEventEngine()->Run(
+          [done, config, i, j] {
```

Comment:
This uses 3 copies of a shared_ptr where I think a single owner would do. For example:

```
size_t size = GPR_ARRAY_SIZE(interesting_sizes);
std::vector<absl::Notification> dones{std::vector<absl::Notification>(size*size)};
for (i ...) {
  for(j ...) {
    Run([&done = dones[i&size + j], &config, i, j] {
      ...
    }
  }
}
```

Reply:
true... but without it we don't need an indexing scheme, so it's kind of brain dead - which I think is the right tradeoff for a test

Reply:
That's a fair argument, especially for tests, I won't block it. But for readability reference, the style guide advises against using shared_ptr unless you have a "very good reason", and we could debate whether avoiding vector index access qualifies as a "very good reason"
 https://google.github.io/styleguide/cppguide.html#Ownership_and_Smart_Pointers.